src/App/Document.cpp

```diff
@@ -2250,6 +2253,56 @@ bool Document::saveToFile(const char* filename) const
     return true;
 }
 
+void Document::registerLabel(const std::string& newLabel)
+{
+    if (newLabel.size() != 0) {
+        if (!d->objectLabelManager.containsName(newLabel)) {
+            // First occurrence of label. We make no entry in objectLabelCounts when the count is one.
+            d->objectLabelManager.addExactName(newLabel);
+        }
+        else {
+            auto it = d->objectLabelCounts.find(newLabel);
+            if (it != d->objectLabelCounts.end()) {
+                // There is a count already greater then one, so increment it
+                it->second++;
+            }
+            else {
+                // There is no count entry, which implies one, so register a count of two
+                d->objectLabelCounts.insert(std::pair(newLabel, 2));
+            }
+        }
+    }
+}
+
+void Document::unregisterLabel(const std::string& oldLabel)
+{
+    if (oldLabel.size() != 0) {
+        auto it = d->objectLabelCounts.find(oldLabel);
+        if (it == d->objectLabelCounts.end()) {
+            // Missing count implies a count of one, or an unregistered name
+            d->objectLabelManager.removeExactName(oldLabel);
+        }
+        else if (--it->second == 1) {
+            // Decremented to one, remove the count entry
+            d->objectLabelCounts.erase(it);
+        }

```

Comment:
It would seem that some find the use of `else if`, especially in a long chain of such, to be hard to understand.
In this case the body of the first `if` is simple so I suppose the suggested change would be just as easy to understand, but this is not in general true. Once the body of the first `if` becomes more complex, it can become more difficult to determine if it always returns, skipping the second `if` whereas the `else if` construct makes this clear without having to examine the body of the first `if` at all. This change also does not work if the entire construct is embedded in other code and one wants execution to always continue after the construct (again, that does not apply here).